azure/services/aso/aso.go

```diff
@@ -155,6 +155,9 @@ func (s *Service) CreateOrUpdateResource(ctx context.Context, spec azure.ASOReso
 
 	if t, ok := spec.(TagsGetterSetter); ok {
 		if err := reconcileTags(t, existing, parameters); err != nil {
+			if azure.IsOperationNotDoneError(err) && readyErr != nil {
+				return nil, readyErr
```

Comment:
This line was the cause for the merge conflict, so this changes `zero` to `nil` from the original PR to account for the generics changes that were introduced for 1.12.